android/coretest/src/androidTest/java/com/mongodb/stitch/android/core/StitchAppClientIntTests.kt

```diff
@@ -32,6 +35,8 @@ import java.util.Arrays
 import java.util.Date
 import java.util.Calendar
 import java.util.concurrent.ExecutionException
+import java.util.concurrent.Semaphore
+import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
 class StitchAppClientIntTests : BaseStitchAndroidIntTest() {
```

Comment:
[task] I think you should probably test more cases auth listener cases. 

check out the JS PR, I added a new file of `StitchAuthListenerIntTests`, and for each method of the listener, I have a test method that simulates the different ways that event can be triggered, and ensures that the event method is called a certain number of times.

it sounds like a lot but it should be pretty simple to write, and it's a lot cleaner than trying to mix in with existing integration tests